deeplearning4j-keras/src/main/java/org/deeplearning4j/keras/DeepLearning4jEntryPoint.java

```diff
@@ -0,0 +1,112 @@
+                multiLayerNetwork.init();
+            } else {
+                throw new RuntimeException("Model type unsupported! (" + entryPointFitParameters.getType() + ")");
+            }
+
+            INDArray features = h5FileToNDArray(entryPointFitParameters.getTrainFeaturesFile());
+            INDArray labels = h5FileToNDArray(entryPointFitParameters.getTrainLabelsFile());
+
+            INDArrayIndex[] ndIndexes = new INDArrayIndex[features.shape().length];
+            for (int i = 0; i < ndIndexes.length; i++) {
+                ndIndexes[i] = NDArrayIndex.all();
+            }
+
+            for (int i = 0; i < entryPointFitParameters.getNbEpoch(); i++) {
+                log.info("Fitting: " + i);
+
+                int begin = 0;
+                while (begin < features.size(0)) {
+                    int end = begin + entryPointFitParameters.getBatchSize();
+
+                    if (log.isTraceEnabled()) {
+                        log.trace("Processing batch: " + begin + " " + end);
+                    }
+
+                    ndIndexes[0] = NDArrayIndex.interval(begin, end);
+                    INDArray featuresBatch = features.get(ndIndexes);
+                    INDArray labelsBatch = labels.get(NDArrayIndex.interval(begin, end));
+                    multiLayerNetwork.fit(featuresBatch, labelsBatch);
+
+                    begin += entryPointFitParameters.getBatchSize();
+                }
+            }
+
+            log.info("Learning model finished");
+        } catch (Throwable e) {
+            log.error("Error while handling request!", e);
+            throw e;
+        }
+    }
+
+    private INDArray h5FileToNDArray(String inputFilePath) {
+        hdf5.H5File h5File = new hdf5.H5File();
+        h5File.openFile(inputFilePath, H5F_ACC_RDONLY);
+        hdf5.DataSet dataSet = h5File.asCommonFG().openDataSet("data");
+        int[] shape = extractShape(dataSet);
+        long totalSize = ArrayUtil.prodLong(shape);
+        float[] dataBuffer = readFromDataSet(dataSet, (int) totalSize);
+
+        INDArray input = Nd4j.create(shape);
+        new RecursiveCopier(input, dataBuffer, shape).copy();
+
+        h5File.close();
+
+        return input;
+    }
+
+    private float[] readFromDataSet(hdf5.DataSet dataSet, int total) {
+        float[] dataBuffer = new float[total];
+        FloatPointer fp = new FloatPointer(dataBuffer);
+        dataSet.read(fp, new hdf5.DataType(hdf5.PredType.NATIVE_FLOAT()));
+        fp.get(dataBuffer);
+        return dataBuffer;
+    }
+
+    private int[] extractShape(hdf5.DataSet dataSet) {
+        hdf5.DataSpace space = dataSet.getSpace();
+        int nbDims = space.getSimpleExtentNdims();
+        long[] shape = new long[nbDims];
+        space.getSimpleExtentDims(shape);
+        return toInt(shape);
+    }
+
+    private int[] toInt(long[] array) {
+        int[] retVal = new int[array.length];
+
+        for (int i = 0; i < array.length; i++) {
```

Comment:
Add this in to ArrayUtil if it's not already. (This is in nd4j where I put this kind of stuff.)